src/Utils.vala

```diff
@@ -264,13 +264,6 @@ namespace Utils {
         string hostname = get_ubiquity_compatible_hostname () ?? ("elementary-os" + "-" + get_chassis ());
         hostname += "-" + get_machine_id ().substring (0, 8);
 
-        // If the automatic hostname logic fails in some way, it's possible we may generate an invalid
-        // hostname. We could fix this by trimming traling/leading hyphens or other invalid characters.
-        // But it's probably a bad hostname anyway, so just fallback
-        if (!Distinst.validate_hostname (hostname)) {
-            hostname = "elementary-os";
-        }
```

Comment:
Did this move somewhere else? It seems like a good thing to have some kind of validation here

Reply:
Since the validation is only available with distinst, I ported the logic with the last commit and took over the tests from distinst.